org.envirocar.app/src/org/envirocar/app/views/login/SigninActivity.java

```diff
@@ -203,19 +207,37 @@ protected void onStart() {
                                     (String) null)
                                     .setCancelable(false)
                                     .show();
+
                         }
+
                     }
 
                     @Override
                     public void onComplete() {
                         if(checkNetworkConnection())
-                        dialog.dismiss();
-                    }
+                            dialog.dismiss();
+
+                        //welcome message after successful login
+                        Snackbar.make(logoImageView, String.format(getResources().getString(
+                                R.string.welcome_message), username), Snackbar.LENGTH_LONG).show();
+                        Handler handler = new Handler();
+                        Runnable x=new Runnable() {
+                            @Override
+                            //finish the activity
+                            public void run() {
+                                finish();
+                            }
+                        };
+                        //post delay after login, 1000ms --> 1sec
+                        // so for around 1 sec welcome message will be shown
+                        handler.postDelayed(x, 1000);
+
```

Comment:
You could rather start an intent for the `BaseMainActivity`, which would be much more straightforward.